### PR TITLE
Announce results of extension search

### LINKED ISSUE #52611: [Accessibility][Usability]A11y_VSCode_ViewExtensions_ScreenReader: NVDA does not announce any information about the data changing dynamically in treeview.
**Environment Details:** 
VSCode Version : 1.24.1
OS Version : Win10

**Additional Details:**
ScreenReader Version: NVDA2018.2

**Repro Steps:**
1. Launch VS Code while NVDA is ON.
2. Navigate to View(Alt+V) menu option and select "Extensions" from dropdown.
3. Type some text in Search Extensions text and check screenreader is reading about it or not.

**Actual:**
NVDA does not announce any information about the data changing dynamically in treeview and also when we click on Clear extensions input button, texts in the textbox gets deleted so this information is also not being read by the screenreader.

**Expected:**
When user types something in the textbox, NVDA should announce some information that how many data been found in the treeview matching those characters. And when user clicks on Delete button, then also screenreader should throw a notification or should announce that data has got deleted.

**Recommendations:**
Refer below link which is repository of bug fixes code snippets:
https://microsoft.sharepoint.com/teams/msenable/mas/pages/browse-fixes.aspx

**User Impact:**
User will not be able to know about the data which are found matching those characters.

Attachment for Reference: 
[A11y VSCode Extensions NVDA clear and insert data.pptx](https://github.com/Microsoft/vscode/files/2123792/A11y.VSCode.Extensions.NVDA.clear.and.insert.data.pptx)

Does this issue occur when all extensions are disabled?: Yes

### PR DESCRIPTION
**Pull request by ramya-rao-a**
This PR fixes #52611 by announcing the number of search results when an extension search is made.

### CODE REVIEW

File: src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts [hunk 1]
```diff
@@ -69,6 +70,18 @@ const SearchBuiltInExtensionsContext = new RawContextKey<boolean>('searchBuiltIn
 const RecommendedExtensionsContext = new RawContextKey<boolean>('recommendedExtensions', false);
 const DefaultRecommendedExtensionsContext = new RawContextKey<boolean>('defaultRecommendedExtensions', false);
 const GroupByServersContext = new RawContextKey<boolean>('groupByServersContext', false);
+const viewIdNameMappings: { [id: string]: string } = {
+	'extensions.listView': localize('marketPlace', "Marketplace"),
```
Comment (octref) [inline src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts]: Use consistent quoting style. For example always prefer single quotes over double quotes.

File: src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts [hunk 2]
```diff
@@ -432,17 +455,36 @@ export class ExtensionsViewlet extends ViewContainerViewlet implements IExtensio
...
+		})));
 		return addedViews;
 	}
 
+	private alertSearchResult(count: number, viewId: string) {
+		if (!count) {
+			return;
+		}
+		if (viewIdNameMappings[viewId]) {
+			alert(localize('extensionsFoundInSection', "{0} extensions found in the {1} section.", count, viewIdNameMappings[viewId]));
```
Comment (octref) [inline src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts]: When `count` is 1, `extensions` is wrong.

File: src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts [hunk 3]
```diff
@@ -432,17 +455,36 @@ export class ExtensionsViewlet extends ViewContainerViewlet implements IExtensio
...
+		} else {
+			alert(localize('extensionsFound', "{0} extensions found.", count));
```
Comment (octref) [inline src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts]: Same as above.

**Review by octref — CHANGES_REQUESTED** [review]

File: src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts (hunk 1 above)
Comment (ramya-rao-a) [reply src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts]: That is true, but not when we localize strings. If we use single quotes on the string that is to be localized, then the localization doesnt work.

We have an exception on our pre check in hook for this. If double quotes were used elsewhere, then I wouldnt have been able to commit in the first place

Comment (octref) [reply src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts]: You can still prefer single quotes first and fallback to backtick if string includes single quotes.

**Review by octref — APPROVED** [review]

File: src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts (hunk 1 above)
Comment (ramya-rao-a) [reply src/vs/workbench/parts/extensions/electron-browser/extensionsViewlet.ts]: The rest of the file is using double quotes, so I'll keep so for consistency's sake

**Pull request merged**